tests/UI/tools/stepIdentifiersChecker.ts

```diff
@@ -0,0 +1,192 @@
+    const testReturned = test;
+    testReturned.file = file;
+
+    return testReturned;
+  });
+
+  const nestedSuites: JsonReportSuite[] = suite.suites;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const nestedSuite of nestedSuites) {
+    tests = tests.concat(getTestsFromSuite(nestedSuite));
+  }
+
+  return tests;
+};
+
+/**
+ * Check for undefined context
+ * @param jsonFile
+ */
+const checkUndefined = (jsonFile: string): boolean => {
+  const allTests = getAllTests(jsonFile);
+  const undefinedContextsSteps: string[] = [];
+  allTests.forEach((value: JsonReportTest): void => {
+    if ((value.context === undefined || !value.context) && !value.pending) {
+      undefinedContextsSteps.push(value.fullTitle.trim());
+    }
+  });
+
+  if (undefinedContextsSteps.length !== 0) {
+    console.error(
+      `Some steps (${undefinedContextsSteps.length}) are missing contexts on these scenarios: \n - ${
+        undefinedContextsSteps.join('\n - ')}`,
+    );
+    return false;
+  }
+  return true;
+};
+
+/**
+ * Check for doubles in mochawesome report
+ * @param {string} jsonFile
+ * @returns {boolean}
+ */
+const checkDoubles = (jsonFile: string): boolean => {
+  const allTests: JsonReportTest[] = getAllTests(jsonFile);
+
+  const reportContexts: string[] = Object.values(allTests)
+    .filter((test: JsonReportTest) => test.context !== undefined && test.context !== null)
+    .map((test: JsonReportTest) => JSON.parse(test.context ?? '').value);
+
+  const contextExisting: string[] = [];
+  const contextDoubles: string[] = [];
+  reportContexts.forEach((value: string): void => {
+    if (['loginBO', 'logoutBO', 'loginFO', 'logoutFO'].indexOf(value) !== -1) {
+      return;
+    }
+    if (contextExisting.indexOf(value) !== -1) {
+      contextDoubles.push(value);
+      return;
+    }
+    contextExisting.push(value);
+  });
+
+  if (contextDoubles.length !== 0) {
+    console.error(`Some test identifiers (${contextDoubles.length}) must be fixed:\n - ${
+      contextDoubles.join('\n - ')}`);
+    return false;
+  }
+  return true;
+};
+
+const checkBaseContext = (jsonFile: string): boolean => {
+  type reportContext = {
+    file: string
+    baseContext: string
+  };
+
+  const allTests: JsonReportTest[] = getAllTests(jsonFile);
+
+  const reportBaseContext: string[] = [];
+  const reportContexts: reportContext[] = Object.values(allTests)
+    .filter((test: JsonReportTest) => test.context)
+    .map((test: JsonReportTest): reportContext|false => {
+      const value: string[] = JSON.parse(test.context ?? '').value.split('_');
+      // Extract the base context from the context
+      const baseContext: string = value.slice(0, value.length - 1).join('_');
+      const {file} = test;
+
+      if (!baseContext || reportBaseContext.includes(file)) {
+        return false;
+      }
+
+      reportBaseContext.push(file);
+      return {file, baseContext};
+    })
+    .filter((test): test is reportContext => test !== false);
```

Comment:
This is a bit rough tbh, testing that `test !== false` doesn't mean that it's a `reportContext`

Reply:
Fixed